src/components/ClaimENSPage/ClaimENSPage.tsx

```diff
@@ -1,16 +1,62 @@
 import * as React from 'react'
-import { Row, Column, Section, Narrow, InputOnChangeData, Header, Form, Field, Button, Mana } from 'decentraland-ui'
+import { Address } from 'web3x-es/address'
+import { Row, Column, Section, Narrow, InputOnChangeData, Header, Form, Field, Button, Mana, Radio } from 'decentraland-ui'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 import { locations } from 'routing/locations'
 import Back from 'components/Back'
 import LoggedInDetailPage from 'components/LoggedInDetailPage'
 import { MAX_NAME_SIZE, isNameValid, PRICE } from 'modules/ens/utils'
 import { Props, State } from './ClaimENSPage.types'
+import { createEth } from 'decentraland-dapps/dist/lib/eth'
+
 import './ClaimENSPage.css'
+import { ERC20 as MANAToken } from 'contracts/ERC20'
+import { CONTROLER_ADDRESS, MANA_ADDRESS } from 'modules/common/contracts'
+import { Eth } from 'web3x-es/eth'
 
 export default class ClaimENSPage extends React.PureComponent<Props, State> {
   state: State = {
-    name: ''
+    name: '',
+    amountApproved: 0,
+    isLoading: false,
+    tx: undefined
+  }
+
+  async componentDidUpdate() {
+    const eth: Eth | null = await createEth()
+    const { address } = this.props
+
+    if (eth && address) {
+      const contractMANA = new MANAToken(eth, Address.fromString(MANA_ADDRESS))
+      const allowance: string = await contractMANA.methods
+        .allowance(Address.fromString(address), Address.fromString(CONTROLER_ADDRESS))
+        .call()
+      const amountApproved: number = +allowance
+      this.setState({ amountApproved })
```

Comment:
You can do this in one line if you want
```suggestion
      this.setState({ amountApproved: +allowance })
```
not trying to say this is strictly worse, but we're saving an unused variable